pkg/worker/k8spodtemplating.go

```diff
@@ -76,24 +66,382 @@ func getPodSpec(ctx context.Context, step wharfyml.Step) (v1.Pod, error) {
 				},
 			},
 		},
-	}, nil
+	}
+
+	if err := applyStep(&pod, step); err != nil {
+		return v1.Pod{}, err
+	}
+
+	if len(pod.Spec.Containers) == 0 {
+		return v1.Pod{}, errors.New("step type did not add an app container")
+	}
+
+	return pod, nil
 }
 
-func getPodImage(step wharfyml.Step) (string, error) {
-	switch s := step.Type.(type) {
-	case wharfyml.StepContainer:
-		return s.Image, nil
-	default:
-		return "", fmt.Errorf("unsupported step type: %q", step.Type)
+func getPodGenerateName(step wharfyml.Step) string {
+	name := fmt.Sprintf("wharf-build-%s-%s-",
+		sanitizePodName(step.Type.StepTypeName()),
+		sanitizePodName(step.Name))
+	// Kubernetes API will respond with error if the GenerateName is too long.
+	// We trim it here to less than the 253 char limit as 253 is an excessive
+	// name length.
+	const maxLen = 42 // jokes aside, 42 is actually a great maximum name length
+	// For reference, this is what a 42-long name looks like:
+	// wharf-build-container-some-long-step-name-
+	if len(name) > maxLen {
+		name = name[:maxLen-1] + "-"
```

Comment:
nit: This feels rather arbitrary.
If a user would want to use long step names for whatever reason, why should we put a hard limit on the length?

Maybe a higher limit or no limit (i.e., matching k8s's limitation of 253) for now, and make it configurable later in a different PR?

Put as nit so feel free to completely ignore and just resolve this comment 👍🏻

Reply:
Because I don't want to do `kubectl get pods` and have Wharf pod names wrap like crazy.
The names doesn't really matter, it's just to give some kind of indignation of what the pod is actually doing when doing `kubectl get pods`.

If you want the full names, then we could add a `wharf-cmd worker list` or something that extracts the full stage and step names from the pod's annotations.

Reply:
I.e, having the step name in there is a QoL feature. We could've just named them `wharf-build-`, which is pretty much what Jenkins does right now.

I do agree that we could make it configurable, but as we don't have that kind of infrastructure for the wharf-cmd project yet, I'm leaving that for a future PR, if ever.